core/src/main/java/org/bitcoinj/core/Utils.java

```diff
@@ -67,12 +67,17 @@
      * @return numBytes byte long array.
      */
     public static byte[] bigIntegerToBytes(BigInteger b, int numBytes) {
-        byte[] bytes = new byte[numBytes];
-        byte[] biBytes = b.toByteArray();
-        int start = (biBytes.length == numBytes + 1) ? 1 : 0;
-        int length = Math.min(biBytes.length, numBytes);
-        System.arraycopy(biBytes, start, bytes, numBytes - length, length);
-        return bytes;        
+        checkArgument(b.signum() > 0, "We assume only positive numbers");
+        checkArgument(numBytes > 0, "numBytes must be positive");
+        byte[] src = b.toByteArray();
+        byte[] dest = new byte[numBytes];
+        byte firstByte = src[0];
```

Comment:
We could even extract the whole expression `src[0] == 0` and give it a meaningful name.